kclvm/sema/src/advanced_resolver/node.rs

```diff
@@ -867,7 +867,24 @@ impl<'ctx> MutSelfTypedResultWalker<'ctx> for AdvancedResolver<'ctx> {
     }
 
     fn walk_config_expr(&mut self, config_expr: &'ctx ast::ConfigExpr) -> Self::Result {
-        self.walk_config_entries(&config_expr.items)?;
+        let (start, _end) = (self.ctx.start_pos.clone(), self.ctx.end_pos.clone());
+
+        let pre_pos = Position {
+            filename: start.filename.clone(),
+            line: start.line.clone(),
+            column: start.column.map(|c| if c >= 1 { c - 1 } else { 0 }),
+        };
+
+        let with_hint = if let Some(stmt) = self.ctx.program.pos_to_stmt(&pre_pos) {
```

Comment:
I guess hints have to applied only for anonymous schema configs, like 

```
schema Name:
    name: str

n: Name = {
    name = "kcl"
}

```
that's why I used assign_stmt to determine if type annotations exist for the same, if yes then hints are applied, otherwise not.

Reply:
![Screenshot from 2024-08-28 18-21-55](https://github.com/user-attachments/assets/4542d981-39db-42f3-bce8-35e581a97898)

hints are being applied for the `config` too

Reply:
![Screenshot from 2024-08-28 18-35-23](https://github.com/user-attachments/assets/aac5b4cc-909f-4abd-8bd5-44bf613b5801)

I guess we need to add check for schema_expr too